templates/single-tapestry.php

```diff
@@ -41,17 +43,28 @@ function addNonceToScript()
                 comments_template();
             endif;
             ?>
+        
+        <?php endwhile; // end of the loop. ?>
 
-        <?php endwhile;
-    ?>
+        <div id="tapestry">
+            <!-- Don't render add root button when there's nodes -->
+            <?php if(!doesTapestryHaveNodes(get_the_ID())) : ?>
```

Comment:
Actually could you move this into `tapestry.js` so it's dynamically added in instead? It's better like that since we're adding everything like that. Thanks!

Reply:
done.